cpp/ql/src/semmle/code/cpp/ir/implementation/aliased_ssa/Instruction.qll

```diff
@@ -321,6 +321,17 @@ class Instruction extends Construction::TInstruction {
     Construction::hasModeledMemoryResult(this)
   }
 
+  /**
+   * Holds if this is an instruction with a memory result that represents a
+   * conflation of more than one object.
+   *
+   * This happens in practice when dereferencing a pointer that cannot be
+   * tracked back to a single local allocation. Such memory is instead modeled
+   * as originating on the `AliasedDefinitionInstruction` at the entry of the
+   * function.
+   */
+  final predicate isResultConflated() { Construction::hasConflatedMemoryResult(this) }
```

Comment:
I modelled the naming of this predicate on `isResultModeled`, which is right above it.

It might be confusing that this predicate is inverted compared to `isResultModeled`: the new predicate holds when the instruction is imprecise, whereas `isResultModeled` holds when it's (in a sense) precise. Do we need `isResultModeled`? It's currently only used for the pretty-printer, and I think `isResultConflated` might be better for pretty-printing.

Reply:
I don't think there will be much use of `isResultModeled`. In fact, I believe that your usage is the first outside of the `Instruction` class itself.

I like having unmodeled results and operands print out as `mu` instead of `m`. I do think that `isResultConflated` is worth displaying in IR dumps as well, though. I'm not sure what punctuation to use for that, but Emoji open up a world of possibilities:
`#  134|     m134_4🔀(unknown)      = Chi                     : total:m134_2, partial:m134_3`

Reply:
Um... yes. I was thinking `ma` for results and operands that are conflated but not unmodelled.

Reply:
...where the `a` is for Aliased.